server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -738,6 +767,20 @@ public void run() {
                 }
             }
         }
+
+        LongSupplier relativeTimeInMillisSupplier() {
```

Comment:
NIT: Kinda weird to have this as a function when we only call it once during the lifetime of any instance, why not just make this a final field that we set up in the constructor of the thread pool? Doesn't really matter much though :)

Reply:
Yeah I can tidy this up; also this area is pretty light on tests right now, I'll add some before marking this as non-draft.